libraries/Microsoft.Bot.Builder.Prompts/OAuthPrompt.cs

```diff
@@ -0,0 +1,173 @@
+            var replyActivity = MessageFactory.Attachment(cardAttachment);
+            await context.SendActivity(replyActivity);
+        }
+
+        public async Task<TokenResult> Recognize(ITurnContext context)
+        {
+            if (IsTokenResponseEvent(context))
+            {
+                var tokenResponse = context.Activity.Value as TokenResponse;
+                return new TokenResult() { Status = PromptStatus.Recognized, Value = tokenResponse };
+
+            }
+            else if (context.Activity.Type == ActivityTypes.Message)
+            {
+                var matched = regex.Match(context.Activity.Text);
+                if (matched.Success)
+                {
+                    var adapter = context.Adapter as BotFrameworkAdapter;
+                    if (adapter == null)
+                        throw new InvalidOperationException("OAuthPrompt.Recognize(): not supported by the current adapter");
+                    var token = await adapter.GetUserToken(context, _settings.ConnectionName, matched.Value);
+                    var tokenResult = new TokenResult() { Status = PromptStatus.Recognized, Value = token };
+                    if (_promptValidator != null)
+                        await _promptValidator(context, tokenResult);
+                    return tokenResult;
+                }
+            }
+            return new TokenResult() { Status = PromptStatus.NotRecognized };
+        }
+
+        public async Task<TokenResult> GetUserToken(ITurnContext context)
+        {
+            var adapter = context.Adapter as BotFrameworkAdapter;
+            if (adapter == null)
+                throw new InvalidOperationException("OAuthPrompt.GetUserToken(): not supported by the current adapter");
+
+            var token = await adapter.GetUserToken(context, _settings.ConnectionName, null);
+            TokenResult tokenResult = null;
+            if (token == null)
+                tokenResult = new TokenResult() { Status = PromptStatus.NotRecognized };
+            else
+                tokenResult = new TokenResult() { Status = PromptStatus.Recognized, Value = token };
+            if (_promptValidator != null)
+                await _promptValidator(context, tokenResult);
+            return tokenResult;
+        }
+
+        public async Task SignOutUser(ITurnContext context)
+        {
+            var adapter = context.Adapter as BotFrameworkAdapter;
+            if (adapter == null)
+                throw new InvalidOperationException("OAuthPrompt.SignOutUser(): not supported by the current adapter");
+
+            // Sign out user
+            await adapter.SignOutUser(context, _settings.ConnectionName);
+        }
+
+        private bool IsTokenResponseEvent(ITurnContext context)
+        {
+            var activity = context.Activity;
+            return (activity.Type == ActivityTypes.Event && activity.Name == "tokens/response");
+        }
+
+        private bool asSignInCard(string channelId)
```

Comment:
The name of this method doesn't match the behavior. 

"DoesChannelSupportOAuth" would be a better name.